plugin/trino-delta-lake/src/main/java/io/trino/plugin/deltalake/transactionlog/TableSnapshot.java

```diff
@@ -199,30 +203,27 @@ public Stream<DeltaLakeTransactionLogEntry> getCheckpointTransactionLogEntries(
 
         return getCheckpointPartPaths(checkpoint).stream()
                 .map(fileSystem::newInputFile)
-                .flatMap(checkpointFile -> {
-                    CheckpointEntryIterator checkpointEntryIterator = getCheckpointTransactionLogEntries(
-                            session,
-                            entryTypes,
-                            metadataAndProtocol.map(MetadataAndProtocolEntry::metadataEntry),
-                            metadataAndProtocol.map(MetadataAndProtocolEntry::protocolEntry),
-                            checkpointSchemaManager,
-                            typeManager,
-                            stats,
-                            checkpoint,
-                            checkpointFile,
-                            partitionConstraint,
-                            addStatsMinMaxColumnFilter);
-                    return Streams.stream(checkpointEntryIterator)
-                            .onClose(checkpointEntryIterator::close);
-                });
+                .flatMap(checkpointFile -> getCheckpointTransactionLogEntries(
+                        session,
+                        entryTypes,
+                        metadataAndProtocol.map(MetadataAndProtocolEntry::metadataEntry),
+                        metadataAndProtocol.map(MetadataAndProtocolEntry::protocolEntry),
+                        checkpointSchemaManager,
+                        typeManager,
+                        stats,
+                        checkpoint,
+                        checkpointFile,
+                        partitionConstraint,
+                        addStatsMinMaxColumnFilter,
+                        fileSystem));
     }
 
     public Optional<Long> getLastCheckpointVersion()
     {
         return lastCheckpoint.map(LastCheckpoint::version);
     }
 
-    private CheckpointEntryIterator getCheckpointTransactionLogEntries(
```

Comment:
nit: this API change can be a separate commit (no change requested)

Reply:
Let me skip this comment as separated commit doesn't improve diff so much.